caldav/objects.py

```diff
@@ -876,23 +899,28 @@ def date_search(
         ## returned.
         objects = self.search(root, comp_class)
         if expand:
+            expanded_objects = []
             for o in objects:
+                has_expanded = False
                 if not o.data:
+                    expanded_objects.append(o)
                     continue
                 components = o.vobject_instance.components()
                 for i in components:
                     if i.name == "VEVENT":
                         recurrance_properties = ["exdate", "exrule", "rdate", "rrule"]
                         if any(key in recurrance_properties for key in i.contents):
-                            if verify_expand:
-                                raise error.ReportError(
-                                    "CalDAV server did not expand recurring vevents as requested. See https://github.com/python-caldav/caldav/issues/157"
-                                )
-                            else:
-                                logging.error(
-                                    "CalDAV server does not support recurring events properly.  See https://github.com/python-caldav/caldav/issues/157"
-                                )
-        return objects
+                            # fall back to next 24 hours if open-ended search
```

Comment:
In general, I'm all for being pragmatic, and in many cases it may be better to have a "sensible default" than raising an error.  I would argue that if an open-ended search is requested, "24 hours" is quite a short timespan, the "sensible default" would probably be weeks, months or maybe even a year.  Anyway, in line 822 we already reject open-ended date searches before sending it to the server, hence I think we should be consistent and reject it here, too.

```
        if not end and expand:
            raise error.ReportError("an open-ended date search cannot be expanded")
```

Reply:
I was not sure myself how to approach this; raising an error seems more sane.